### PR TITLE
Event onReady

### PR DESCRIPTION
**Pull request by frenicohansen**
### Background

As I was working with the library, I encountered a function called onPositionComplete. According to its documentation, this function was meant to "add a listener used to execute a function once all items in the grid have been properly positioned." However, something about the name and the description felt off. It gave the impression that the listener would only be triggered a single time, likely on the first positioning of the items in the grid.

As I dug deeper into the code, it became clear that onPositionComplete actually behaved differently than what the name and documentation suggested. The listener wasn't just a one-time thing—it would trigger every time the positionItems() method was called or when the container was resized. This behavior was useful, but the existing naming and documentation didn't make that clear.

### Changes
- Introduced `onReady` to run a listener only once, after the first positioning of items.
  - This is useful for scenarios like managing a loading state, where you only want to take action once the grid is properly positioned for the first time, as mentioned in #56.
- Fixed the return type of the original onPositionComplete from void to number to match its actual behavior.
- Updated documentation for onPositionComplete and onGridReady

### CODE REVIEW

**Comment by e-oj** [comment]
Nice idea! I think we can just emit a `ready` event in the listen function, add an `onReady` function, and leave everything else the same. That way there won't be any breaking changes with this feature. 

The current setup, where `positionComplete` is emitted whenever `positionItems` is called, makes sense to me so I don't think we have to add an `onRepositionComplete`

**Comment by e-oj** [comment]
Also you're right about the documentation; we should update that to make it clearer as well. Thanks for pointing this out!

**Comment by frenicohansen** [comment]
> Nice idea! I think we can just emit a `ready` event in the listen function, add an `onReady` function, and leave everything else the same. That way there won't be any breaking changes with this feature.
> 
> The current setup, where `positionComplete` is emitted whenever `positionItems` is called, makes sense to me so I don't think we have to add an `onRepositionComplete`

I updated the PR description and implemented your suggestion. With this there should be no breaking changes.

File: index.d.ts [hunk 1]
```diff
@@ -165,11 +165,17 @@ declare class MagicGrid extends EventEmitter {
...
+    * @param callback - function to be called on grid ready
+    */
+    onReady(callback: () => void): number;
+
     /**
      * Adds a callback for when positioning is complete
      * @param callback - function to be called on positioning complete
      */
-    onPositionComplete(callback: () => void): void;
+    onPositionComplete(callback: () => void): number;
```
Comment (e-oj) [inline index.d.ts:178]: Nice catch!

File: README.md [hunk 2]
```diff
@@ -163,8 +163,27 @@ magicGrid.positionItems();
...
+});
+
+magicGrid.listen();
+```
+
+---
+
 #### .onPositionComplete(callback)
-Adds a listener used execute a function once all items in the grid have been properly positioned.
+Adds a listener that executes a function once the `positionItems()` method has been called or after the container has been resized.
```
Comment (e-oj) [inline README.md:186]:
```suggestion
Adds a listener that executes a function whenever `positionItems` is called. 
**Note: `positionItems` is called during initial setup and whenever the container or window is resized**
```

**Review by e-oj — APPROVED** [review]

**Comment by e-oj** [comment]
Awesome work. Thanks!

**Pull request merged**